pkg/build/api/validation/validation.go

```diff
@@ -445,6 +462,28 @@ func validateCustomStrategy(strategy *buildapi.CustomBuildStrategy, fldPath *fie
 	return allErrs
 }
 
+func validateJenkinsPipelineStrategy(strategy *buildapi.JenkinsPipelineBuildStrategy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(strategy.JenkinsfilePath) != 0 && len(strategy.Jenkinsfile) != 0 {
+		return append(allErrs, field.Invalid(fldPath, strategy, "must provide a value for exactly one of jenkinsfilePath, or jenkinsfile"))
+	}
+	if len(strategy.JenkinsfilePath) != 0 {
+		cleaned := path.Clean(strategy.JenkinsfilePath)
+		switch {
+		case strings.HasPrefix(cleaned, "/"):
```

Comment:
Agreed - this was actually copied from [Docker build validation](https://github.com/openshift/origin/blob/e91867dadcb7e014dcf73d944f1864307a884307/pkg/build/api/validation/validation.go#L415) so someone should probably update that too.